src/graphics/resolution.h

```diff
@@ -0,0 +1,81 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  A list common video resolutions.
+ */
+
+#ifndef GRAPHICS_RESOLUTION_H
+#define GRAPHICS_RESOLUTION_H
+
+namespace Graphics {
+
+struct Resolution {
+	int width;
+	int height;
+
+	Resolution(int w, int h) : width(w), height(h) { }
+};
+
+static std::vector<Resolution> initResolutions() {
+	// Add all standard resolutions to the list
+	std::vector<Resolution> resolutions;
+	resolutions.reserve(33);
+	resolutions.push_back(Resolution(7680, 4800));
+	resolutions.push_back(Resolution(7680, 4320));
+	resolutions.push_back(Resolution(6400, 4800));
+	resolutions.push_back(Resolution(6400, 4096));
+	resolutions.push_back(Resolution(5120, 4096));
+	resolutions.push_back(Resolution(5120, 3200));
+	resolutions.push_back(Resolution(4096, 3072));
+	resolutions.push_back(Resolution(4096, 1716));
+	resolutions.push_back(Resolution(3840, 2400));
+	resolutions.push_back(Resolution(3200, 2400));
+	resolutions.push_back(Resolution(3200, 2048));
+	resolutions.push_back(Resolution(2560, 2048));
+	resolutions.push_back(Resolution(2560, 1600));
+	resolutions.push_back(Resolution(2560, 1440));
+	resolutions.push_back(Resolution(2048, 1536));
+	resolutions.push_back(Resolution(2048, 1152));
+	resolutions.push_back(Resolution(2048, 1080));
+	resolutions.push_back(Resolution(1920, 1200));
+	resolutions.push_back(Resolution(1920, 1080));
+	resolutions.push_back(Resolution(1680, 1050));
+	resolutions.push_back(Resolution(1600, 1200));
+	resolutions.push_back(Resolution(1600,  900));
+	resolutions.push_back(Resolution(1440,  900));
+	resolutions.push_back(Resolution(1400, 1050));
+	resolutions.push_back(Resolution(1280, 1024));
+	resolutions.push_back(Resolution(1280,  800));
+	resolutions.push_back(Resolution(1280,  720));
+	resolutions.push_back(Resolution(1152,  864));
+	resolutions.push_back(Resolution(1024,  768));
+	resolutions.push_back(Resolution(800 ,  600));
+	resolutions.push_back(Resolution(640 ,  480));
+	resolutions.push_back(Resolution(320 ,  240));
+	resolutions.push_back(Resolution(320 ,  200));
+	return resolutions;
+}
+
+static std::vector<Resolution> resolutions = initResolutions();
```

Comment:
Well, it would still be allocated on the heap and during the init code vs. being pulled into static memory wholesale with the data section; and it adds to the global constructors. It's simply unnecessary.